frame/staking/src/migrations.rs

```diff
@@ -287,217 +291,228 @@ pub mod v10 {
+				let prev_count = T::VoterList::count();
+				let weight_of_cached = Pallet::<T::Config>::weight_of_fn();
+				for (v, _) in Validators::<T::Config>::iter() {
+					let weight = weight_of_cached(&v);
+					let _ = T::VoterList::on_insert(v.clone(), weight).map_err(|err| {
+						frame_support::log::warn!("failed to insert {:?} into VoterList: {:?}", v, err)
+					});
+				}
+
+				frame_support::log::info!(
+					"injected a total of {} new voters, prev count: {} next count: {}, updating to version 9",
+					Validators::<T::Config>::count(),
+					prev_count,
+					T::VoterList::count(),
+				);
+
+				obsolete::StorageVersion::<T::Config>::put(obsolete::Releases::V9_0_0);
+				<T::Config as frame_system::Config>::BlockWeights::get().max_block
+			} else {
+				frame_support::log::warn!(
+					"InjectValidatorsIntoVoterList being executed on the wrong storage \
+				version, expected obsolete::Releases::V8_0_0"
+				);
+				<T::Config as frame_system::Config>::DbWeight::get().reads(1)
+			}
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+			frame_support::ensure!(
+				obsolete::StorageVersion::<T>::get() == obsolete::Releases::V8_0_0,
+				"must upgrade linearly"
+			);
+
+			let prev_count = T::VoterList::count();
+			Ok(prev_count.encode())
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade(prev_count: Vec<u8>) -> Result<(), &'static str> {
+			let prev_count: u32 = Decode::decode(&mut prev_count.as_slice()).expect(
+				"the state parameter should be something that was generated by pre_upgrade",
+			);
+			let post_count = T::VoterList::count();
+			let validators = Validators::<T>::count();
+			assert!(post_count == prev_count + validators);
+
+			frame_support::ensure!(
+				obsolete::StorageVersion::<T>::get() == obsolete::Releases::V9_0_0,
+				"must upgrade "
+			);
+			Ok(())
+		}
+	}
+}
+
+pub mod v8 {
+	use super::*;
+	use crate::{Config, Nominators, Pallet, Weight};
+	use frame_election_provider_support::{SortedListProvider, ReadOnlySortedListProvider};
+	use frame_support::traits::Get;
+
+	pub trait MigrationConfig {
+		type Config: Config;
+		type VoterList: frame_election_provider_support::SortedListProvider<
+			<Self::Config as frame_system::Config>::AccountId,
+			Score = u64
+		>;
+	}
+
+	#[cfg(feature = "try-runtime")]
+	pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
+		frame_support::ensure!(
+			obsolete::StorageVersion::<T>::get() == obsolete::Releases::V7_0_0,
+			"must upgrade linearly"
+		);
+
+		crate::log!(info, "👜 staking bags-list migration passes PRE migrate checks ✅",);
+		Ok(())
+	}
+
+	/// Migration to sorted `VoterList`.
+	pub fn migrate<T: MigrationConfig>() -> Weight {
+		if obsolete::StorageVersion::<T::Config>::get() == obsolete::Releases::V7_0_0 {
+			frame_support::log::info!("migrating staking to obsolete::Releases::V8_0_0");
+
+			let migrated = T::VoterList::unsafe_regenerate(
```

Comment:
This does is not currently accessible either, but we can make it accessible via RO list, since it's `unsafe`.

Reply:
@kianenigma ^

Reply:
disregard